script.js

```diff
@@ -1,55 +1,101 @@
-function Animal(name){
+class Animal {
+  constructor(name) {
     this.name = name;
-}
-
-Animal.prototype.action = function(){
-  document.getElementById(this.actionSoundName).play();
-  
-}
+  }
 
-Animal.prototype.putInTheDocument = function(){
+  putInTheDocument() {
     var petsTable = document.getElementById("petsTable");
     var petTR = document.createElement("tr");
-  
+
     var petNameTD = document.createElement("td");
     petNameTD.textContent = this.name;
     petTR.appendChild(petNameTD);
-  
+
     var petLegsTD = document.createElement("td");
     petLegsTD.textContent = this.legs;
     petTR.appendChild(petLegsTD);
-  
+
     var petActionTD = document.createElement("td");
     var petActionTDButton = document.createElement("button");
     petActionTDButton.textContent = this.actionText;
     petActionTD.appendChild(petActionTDButton);
     petTR.appendChild(petActionTD);
-    
+
     petActionTDButton.onclick = this.action.bind(this);
-    petsTable.querySelector("tbody").appendChild(petTR)
-}
+    petsTable.querySelector("tbody").appendChild(petTR);
+  }
 
+  action() {
```

Comment:
`   action(e) {
        e.stopPropagation();
        document.getElementById(this.actionSoundName).play();
    }`
    
    ekleyerek Event Bubbling'i engelleyebilirsin. Böylece button'a basıldığında resim gösterilmemiş olur.